### PR TITLE
Adding more world API methods

### LINKED ISSUE #5116: API to /locatebiome programmatically
**Is your feature request related to a problem? Please describe.**
Not necessarily a problem but a headache saver. 

**Describe the solution you'd like**
Basically just a way to locatebiome without needing NMS.

**Describe alternatives you've considered**
pain, agony even.

**Additional context**
Apparently, it's trivial.

### PR DESCRIPTION
**Pull request by Machine-Maker**
Adds api getters for [these world properties](https://minecraft.gamepedia.com/Custom#JSON_format).

Closes #5116

### CODE REVIEW

File: Spigot-API-Patches/0218-More-World-API.patch
Comment (FranKusmiruk) [inline Spigot-API-Patches/0218-More-World-API.patch]:
```suggestion
+    boolean doesBedWork();
```

File: Spigot-API-Patches/0218-More-World-API.patch
Comment (FranKusmiruk) [inline Spigot-API-Patches/0218-More-World-API.patch]:
```suggestion
+    boolean doesRespawnAnchorWork();
```

**Review by FranKusmiruk — COMMENTED** [review]
Just wondering but are these fine without javadoc?

**Comment by Machine-Maker** [comment]
> Just wondering but are these fine without javadoc?

totally forgot... lol

File: Spigot-API-Patches/0218-More-World-API.patch
Comment (Proximyst) [inline Spigot-API-Patches/0218-More-World-API.patch]: Drop this

File: Spigot-API-Patches/0218-More-World-API.patch
Comment (Proximyst) [inline Spigot-API-Patches/0218-More-World-API.patch]:
```suggestion
+     * Locates the nearest biome based on an origin, biome type, and radius to search.
+     * Step defaults to {@code 8}.
```

File: Spigot-API-Patches/0218-More-World-API.patch
Comment (Proximyst) [inline Spigot-API-Patches/0218-More-World-API.patch]:
```suggestion
+     * Checks if the world:
+     * <ul>
+     *     <li>evaporates water</li>
+     *     <li>dries sponges</li>
+     *     <li>has lava spread faster and further</li>
+     * </ul>
+     *
+     * @return whether the world is ultrawarm
```

File: Spigot-API-Patches/0218-More-World-API.patch
Comment (Proximyst) [inline Spigot-API-Patches/0218-More-World-API.patch]:
```suggestion
+     * Checks if the world is natural.
+    * <p>
+     * If {@code false}, compasses will spin randomly in the world.
+     * If {@code true}, nether portals will spawn Zombified Piglins.
+     *
+     * @return whether the world is natural
```

File: Spigot-API-Patches/0218-More-World-API.patch
Comment (Proximyst) [inline Spigot-API-Patches/0218-More-World-API.patch]:
```suggestion
+     * Checks if the world is shrunk where 1 block is 8 blocks in the overworld like the Nether.
```

File: Spigot-API-Patches/0218-More-World-API.patch
Comment (Proximyst) [inline Spigot-API-Patches/0218-More-World-API.patch]:
```suggestion
+     * @return whether the world is shrunk
```

File: Spigot-API-Patches/0218-More-World-API.patch
Comment (Proximyst) [inline Spigot-API-Patches/0218-More-World-API.patch]:
```suggestion
+     * @return whether there is skylight
```

File: Spigot-API-Patches/0218-More-World-API.patch
Comment (Proximyst) [inline Spigot-API-Patches/0218-More-World-API.patch]:
```suggestion
+     * @return whether the world has a bedrock ceiling
```

File: Spigot-API-Patches/0218-More-World-API.patch
Comment (Proximyst) [inline Spigot-API-Patches/0218-More-World-API.patch]:
```suggestion
+    boolean hasBedrockCeiling();
```

File: Spigot-API-Patches/0218-More-World-API.patch
Comment (Proximyst) [inline Spigot-API-Patches/0218-More-World-API.patch]:
```suggestion
+     * Checks if Piglins will convert to Zombified Piglins in this world.
```

File: Spigot-API-Patches/0218-More-World-API.patch
Comment (Proximyst) [inline Spigot-API-Patches/0218-More-World-API.patch]:
```suggestion
+     * @return whether Piglins will <i>not</i> transform
```

File: Spigot-API-Patches/0218-More-World-API.patch
Comment (Proximyst) [inline Spigot-API-Patches/0218-More-World-API.patch]:
```suggestion
+     * @return whether beds work
```

What if they don't work?

File: Spigot-API-Patches/0218-More-World-API.patch
Comment (Proximyst) [inline Spigot-API-Patches/0218-More-World-API.patch]:
```suggestion
+     * @return whether respawn anchors work
```

What if they don't work?

File: Spigot-API-Patches/0218-More-World-API.patch
Comment (Proximyst) [inline Spigot-API-Patches/0218-More-World-API.patch]:
```suggestion
+     * @return whether this world supports raids
```

File: Spigot-API-Patches/0218-More-World-API.patch
Comment (Proximyst) [inline Spigot-API-Patches/0218-More-World-API.patch]:
```suggestion
+     * @return whether the world has fixed time
```

File: Spigot-API-Patches/0218-More-World-API.patch
Comment (Proximyst) [inline Spigot-API-Patches/0218-More-World-API.patch]: This feels weird, as every other method is not inverse. Might be good to rename this?

File: Spigot-API-Patches/0218-More-World-API.patch
Comment (Proximyst) [inline Spigot-API-Patches/0218-More-World-API.patch]:
```suggestion
+     * Gets the collection of materials that burn infinitely in this world.
+     *
+     * @return the materials that will forever stay lit by fire
```

File: Spigot-API-Patches/0218-More-World-API.patch
Comment (Proximyst) [inline Spigot-API-Patches/0218-More-World-API.patch]: Components

File: Spigot-API-Patches/0218-More-World-API.patch
Comment (Proximyst) [inline Spigot-API-Patches/0218-More-World-API.patch]: Components

File: Spigot-Server-Patches/0542-More-World-API.patch
Comment (Proximyst) [inline Spigot-Server-Patches/0542-More-World-API.patch]: OBFHELPER

File: Spigot-Server-Patches/0542-More-World-API.patch
Comment (Proximyst) [inline Spigot-Server-Patches/0542-More-World-API.patch]: OBFHELPER

**Review by Proximyst — DISMISSED** [review]

File: Spigot-API-Patches/0218-More-World-API.patch
Comment (Machine-Maker) [reply Spigot-API-Patches/0218-More-World-API.patch]: I was just taking them directly as they were laid out in DimensionManager

File: Spigot-API-Patches/0236-More-World-API.patch
Comment (kashike) [inline Spigot-API-Patches/0236-More-World-API.patch]: Remove this method from this pull request - tracking in #4830.

File: Spigot-API-Patches/0236-More-World-API.patch
Comment (kashike) [inline Spigot-API-Patches/0236-More-World-API.patch]: Remove this method from this pull request - tracking in #4830.

**Review by kashike — CHANGES_REQUESTED** [review]

File: Spigot-API-Patches/0236-More-World-API.patch
Comment (Machine-Maker) [reply Spigot-API-Patches/0236-More-World-API.patch]: Done

File: Spigot-API-Patches/0237-More-World-API.patch
Comment (kashike) [inline Spigot-API-Patches/0237-More-World-API.patch]: These two methods will end up deprecated as part of #4842 - is it worth even adding them now?

**Review by kashike — DISMISSED** [review]

File: Spigot-API-Patches/0237-More-World-API.patch
Comment (Machine-Maker) [reply Spigot-API-Patches/0237-More-World-API.patch]: Yeah, you're right. I'll just leave a note to add the Component method after adventure get merged

**Comment by Machine-Maker** [comment]
Ok, I removed the broadcastMessage methods that were holding this PR up. Should be good to go now.

**Review by electronicboy — APPROVED** [review]

**Review by electronicboy — APPROVED** [review]

**Comment by Machine-Maker** [comment]
`Rebased after the NMS repackage`

File: Spigot-API-Patches/0283-More-World-API.patch
Comment (MiniDigger) [inline Spigot-API-Patches/0283-More-World-API.patch]:
```suggestion
+     *      if {@code true}, nether portals will spawn zombified piglins.
```

**Review by MiniDigger — APPROVED** [review]
nitpick

**Review by kennytv — APPROVED** [review]

**Pull request merged**
